lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,84 @@
+return if ENV['skip_autoinject']
+
+begin
+  require 'open3'
+
+  _, status = Open3.capture2e({'skip_autoinject' => 'true'}, 'bundle show ddtrace')
+
+  if status.success?
+    puts '[DATADOG LIB INJECTION] ddtrace already installed... skipping auto-injection'
+    return
+  end
+
+  require 'bundler'
+  require 'shellwords'
+
+  if Bundler.frozen_bundle?
+    puts "[DATADOG LIB INJECTION] Cannot inject with frozen Bundler"
+    return
+  end
+
+  version = "<DD_TRACE_VERSION_TO_BE_REPLACED>"
+  sha = "<DD_TRACE_SHA_TO_BE_REPLACED>"
+
+  bundle_add_ddtrace_cmd =
+    if !version.empty?
+      # For public release
+      "bundle add ddtrace --require ddtrace/auto_instrument --version #{version.gsub(/^v/, '').shellescape}"
+    elsif !sha.empty?
+      # For internal testing
+      "bundle add ddtrace --require ddtrace/auto_instrument --github datadog/dd-trace-rb --ref #{sha.shellescape}"
+    end
+
+  unless bundle_add_ddtrace_cmd
+    puts "[DATADOG LIB INJECTION] NO VERSION"
+    return
+  end
+
+  puts "[DATADOG LIB INJECTION] ddtrace is not installed... Perform lib injection for dd-trace-rb."
+
+  gemfile   = Bundler::SharedHelpers.default_gemfile
+  lockfile  = Bundler::SharedHelpers.default_lockfile
+
+  if gemfile.basename.to_s == 'gems.rb'
+    datadog_gemfile = gemfile.dirname + "datadog-Gemfile"
+    datadog_lockfile = lockfile.dirname + "datadog-Gemfile.lock"
+  else
+    datadog_gemfile = gemfile.dirname + ("datadog-#{gemfile.basename}")
+    datadog_lockfile = lockfile.dirname + ("datadog-#{lockfile.basename}")
+  end
```

Comment:
Minor: Consider using the same version of the datadog gemfile for both cases -- after all, we're going to tell bundler to use a specific file, so I don't think there's any reason to name it `datadog-Gemfile` or `datadog-gems.rb` depending on the case.

Consider always calling it `datadog-gems.rb` (or `datadog-Gemfile`).